### ISSUE TITLE
Moves point to top of screen each scroll

### ISSUE THREAD

**Issue by fice-t** [zk-phi/sublimity#32]
Whenever you scroll, the point jumps back to the first visible position in the buffer temporarily before going back to the right position.

Hopefully this isn't necessary?

**Comment by zk-phi** [comment]
do you mean

````
(goto-char (window-start))
````

at the line 53 in sublimity-scroll.el right ?

**Comment by fice-t** [comment]
It seems to be that function. Though even changed to this, it still does it for some reason:
```elisp
(defun sublimity-scroll--vscroll (lins)
  (let ((line-start (line-number-at-pos (window-start))))
    (forward-line (+ (- line-start (line-number-at-pos))
                     lins))
    (set-window-start nil (point))))

**Comment by zk-phi** [comment]
the snippet no longer exists in the recent versions I think.

https://github.com/zk-phi/sublimity/blob/master/sublimity-scroll.el

I suppose that you're using an old one.

**Comment by zk-phi** [comment]
oops, sorry, I misunderstood your comment

**Comment by zk-phi** [comment]
"window-start" is the first position and "point" is the cursor position, thus the line

````
(set-window-start nil (point))
````

sets the cursor to the first visible position (and in addition, line-number-at-pos is not very fast function (see http://emacs.stackexchange.com/questions/3821/a-faster-method-to-obtain-line-number-at-pos-in-large-buffers)).

anyway It's probably not impossible to (visually) stay the cursor while scrolling at the same position, by computing the right position for each step, but more calculation leads less smoothness of the animation and it's not worth implementing I think.

**Comment by fice-t** [comment]
I appear to have found a solution using `internal-show-cursor`. I can't say I understand exactly how it works, but it appears to have solved the by temporarily not showing the cursor:

```elisp
(defun sublimity-scroll--vscroll (lins)
  "FOR ANIMATION USE ONLY"
  (internal-show-cursor nil nil)
  (goto-char (window-start))
  (forward-line lins)
  (set-window-start nil (point)))

(defun sublimity-scroll--post-vscroll (lins)
  (sublimity-scroll--vscroll-effect lins)
  (internal-show-cursor nil t))
```

**Comment by zk-phi** [comment]
hiding the cursor seems reasonable, I'll try
the idea later, thanks!

**Comment by zk-phi** [comment]
implemented in f692af1ba045146f568ee36009a78b79e9cfe21f
